### PR TITLE
feat(*)!: prevent querying inventory for new heartbeat

### PR DESCRIPTION
**Pull request by brooksmtownsend**
## Feature or Problem
This PR sets up the HostHeartbeat struct that we use in Wadm to be able to deserialize either the wasmCloud v0.81.0 version of the heartbeat or the v0.82.0 version of the heartbeat. Once v0.82.0 is released, we can remove the code branches handling the v81 heartbeat and take advantage of the additional information there to prevent querying host inventory on each heartbeat.

## Related Issues
Handles https://github.com/wasmCloud/wasmCloud/pull/1160
Fixes https://github.com/wasmCloud/wadm/issues/191

## Release Information
v0.10.0

## Consumer Impact
Backwards compatible with older versions of the host.

### Unit Test(s)
Some unit tests were modified to rip out old inventory format code. Additionally, tests were modified to support wash 0.25.0 and some new control interface command formats.

### Acceptance or Integration
All tests were updated to wash 0.25.0, wasmcloud 0.81.0, and NATS 2.10.7

### CODE REVIEW

File: test/data/events.json [hunk 1]
```diff
@@ -1,336 +1,382 @@
...
-            "link_name": "default",
-            "public_key": "VAG3QITQQ2ODAOWB5TTQSDJ53XK3SHBEIFNK4AYJ5RKAX2UNSCAPHA5M"
-        },
-        "datacontenttype": "application/json",
-        "id": "5613ee03-8645-4ad8-a3aa-3428a33b7e96",
-        "source": "NB6PMW4RGLBP3NAVUVO2IH34VFJFSX7LF7TJOQCDU4GGUGF3P57SZLPX",
-        "specversion": "1.0",
-        "time": "2023-02-14T19:21:09.011785Z",
-        "type": "com.wasmcloud.lattice.provider_started"
+  {
```
Comment (brooksmtownsend) [inline test/data/events.json:2]: Note that most of this file isn't actually changed, it's just me adding an additional host heartbeat for the v82 format. Formatting changes other than that.

**Comment by brooksmtownsend** [comment]
Tests pass, but I'm still working through one or two things with the heartbeat handling.

File: test/docker-compose-e2e.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ services:
       - 4222:4222
   # Have hosts in 3 different "regions"
   wasmcloud_east:
-    image: wasmcloud/wasmcloud:0.79.0
+    image: wasmcloud/wasmcloud:0.81.0
```
Comment (connorsmith256) [inline test/docker-compose-e2e.yaml:9]: Is it worth testing a mix of 0.81 and 0.82 hosts for coverage of both heartbeat payload structures simultaneously?

**Review by connorsmith256 — APPROVED** [review]

File: test/docker-compose-e2e.yaml (hunk 2 above)
Comment (connorsmith256) [reply test/docker-compose-e2e.yaml:9]: Oh, it might be, but 0.82 doesn't exist yet 🙃

**Pull request merged**